File: java/src/ID_001/Main.java
package ID_001;

import java.util.Scanner;

/**
 * @class ID_001
 */
public class Main {

	/**
	 * @param args
	 */
	public static void main(String[] args) {

		Scanner scan = new Scanner(System.in);
		while (scan.hasNextLine()) {
			String theLine = scan.nextLine();
			if (theLine.equals("")) {
				break;
			}
			Scanner scanner = new Scanner(theLine);
			while (scanner.hasNext()) {
				System.out.println(sum(scanner.nextInt()));
			}
		}
	}

	/**
	 * Sums the natural numbers that are factors of 3 or 5
	 * 
	 * @param maxValue
	 *            number of numbers
	 * @return sum sum of numbers less than maxValue
	 */
	public static int sum(int maxValue) {
		int sum = 0;

		for (int i = 1; i < maxValue; i++) {

			if (i % 5 == 0 || i % 3 == 0) {
				sum = sum + i;
			}
		}
		return sum;
	}
}

File: ruby/src/ID_001/Main.rb
 #Sums the natural numbers that are factors of 3 or 5
 # 
 # @param max
 #   number of numbers
 # @return sum sum of numbers less than max

max = gets.to_i
i = 1
sum = 0
while i < max do 
	if (i % 5 == 0 || i % 3 ==0)then
		sum = sum + i;
	end
	i = i + 1
end
puts sum